src/sinks/gcp/stackdriver/metrics/request_builder.rs

```diff
@@ -0,0 +1,136 @@
+    type Payload = Bytes;
+    type Request = HttpRequest;
+    type Error = io::Error;
+
+    fn compression(&self) -> Compression {
+        Compression::None
+    }
+
+    fn encoder(&self) -> &Self::Encoder {
+        &self.encoder
+    }
+
+    fn split_input(
+        &self,
+        mut events: Vec<Metric>,
+    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
+        let finalizers = events.take_finalizers();
+        let builder = RequestMetadataBuilder::from_events(&events);
+        (finalizers, builder, events)
+    }
+
+    fn build_request(
+        &self,
+        metadata: Self::Metadata,
+        request_metadata: RequestMetadata,
+        payload: EncodeResult<Self::Payload>,
+    ) -> Self::Request {
+        HttpRequest::new(payload.into_payload(), metadata, request_metadata)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct StackdriverMetricsEncoder {
+    pub(super) default_namespace: String,
+    pub(super) started: chrono::DateTime<Utc>,
+    pub(super) resource: gcp::GcpTypedResource,
+}
+
+impl encoding::Encoder<Vec<Metric>> for StackdriverMetricsEncoder {
+    /// Create the object defined [here][api_docs].
+    ///
+    /// [api_docs]: https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.timeSeries/create
+    fn encode_input(
+        &self,
+        input: Vec<Metric>,
+        writer: &mut dyn io::Write,
+    ) -> io::Result<(usize, GroupedCountByteSize)> {
+        let mut byte_size = telemetry().create_request_count_byte_size();
+        let time_series = input
+            .into_iter()
+            .map(|metric| {
+                byte_size.add_event(&metric, metric.estimated_json_encoded_size_of());
+
+                let (series, data, _metadata) = metric.into_parts();
+                let namespace = series
+                    .name
+                    .namespace
+                    .unwrap_or_else(|| self.default_namespace.clone());
+                let metric_type = format!(
+                    "custom.googleapis.com/{}/metrics/{}",
+                    namespace, series.name.name
+                );
+
+                let end_time = data.time.timestamp.unwrap_or_else(chrono::Utc::now);
+
+                let (point_value, interval, metric_kind) = match &data.value {
+                    MetricValue::Counter { value } => {
+                        let interval = gcp::GcpInterval {
+                            start_time: Some(self.started),
+                            end_time,
+                        };
+
+                        (*value, interval, gcp::GcpMetricKind::Cumulative)
+                    }
+                    MetricValue::Gauge { value } => {
+                        let interval = gcp::GcpInterval {
+                            start_time: None,
+                            end_time,
+                        };
+
+                        (*value, interval, gcp::GcpMetricKind::Gauge)
+                    }
+                    _ => unreachable!(),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏I think it'd be helpful to add a small comment saying that the other metric types are filtered out at the beginning of the sink's stream processing.